### PR TITLE
Application: Polkadot Runtime Releaser

### PR DESCRIPTION
**Pull request by AurevoirXavier**
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (DOT, USDC & USDT) address in the application and bank details via email, if applicable).
- [ ] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
I have read and hereby sign the Contributor License Agreement.

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Hi @AurevoirXavier thanks for the application. We don't offer payments on ETH mainnet anymore, can you change the payment address to accept USDC on Asset Hub instead? Thanks!

**Comment by AurevoirXavier** [comment]
> Hi @AurevoirXavier thanks for the application. We don't offer payments on ETH mainnet anymore, can you change the payment address to accept USDC on Asset Hub instead? Thanks!

Done. :)

**Comment by keeganquigley** [comment]
Thanks @AurevoirXavier I will mark the application as ready for review so the committee can comment. In the meantime, one further question:

Your previous grant deliveries were highly rated, but the repos appear to be no longer maintained. What assurances can you give us that you will keep improving and maintaining the CLI tool after the grant milestone is complete? Also can you give an update on the status of Subalfred?

**Comment by AurevoirXavier** [comment]
> Your previous grant deliveries were highly rated, but the repos appear to be no longer maintained. What assurances can you give us that you will keep improving and maintaining the CLI tool after the grant milestone is complete? Also can you give an update on the status of Subalfred?

You can verify that I've made significant improvements since the last git commit. At present, the project is not in active development, but all existing features are functioning well, and I personally use it daily. I've also attempted to recommend this useful tool to others, but it seems I'm not getting enough attention. Perhaps a [co-promotion](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md) would be beneficial. I haven't received enough feedback, and since I focused on my job last year, I haven't added many new features to Subalfred.

Also, I believe one reason is that it's a CLI tool. Many users prefer a straightforward website where they can click a few buttons to achieve their desired results rather than installing a CLI tool. For instance, obtaining a twoxxhash result. I do have plans to apply for a maintenance grant to develop SubalfredV2 and transition it to the web.

**Review by Noc2 — DISMISSED** [review]
Thanks for the application. Given your previous work, I'm happy to support it.

**Review by keeganquigley — APPROVED** [review]
Thanks for your answers @AurevoirXavier I'm also happy to approve it.

We recently started doing KYC/KYB for grantees. While waiting for the third approval, could you please go through the [verification process](https://in.sumsub.com/idensic/l/#/uni_iW29BlDPIhrrmtck) for hack-ink? Thanks!

Let me know if you have any issues during the process.

**Comment by AurevoirXavier** [comment]
> Let me know if you have any issues during the process.

Thank you for the info.

As I mentioned in the application README.

```
### Legal Structure

Personal. (no legal structure entry)
```

hack-ink doesn't really be registered as a real company(in legal). I host all the code source under the hack-ink GitHub organization. So I use it as the Team name.

Then my question is what company document show I upload for this? Should I change to a personal KYC and how?

**Comment by AurevoirXavier** [comment]
> Hi @AurevoirXavier thanks for the explanation, in that case, you can just do [KYC verification](https://in.sumsub.com/idensic/l/#/uni_mjcB05gJZfh8dbr7). Thanks.

I can't see the error on my side. And I can't try it again. Should I use a new email address to try?
```
Unfortunately, we couldn't verify your identity
We could not verify your profile. If you have any questions, please contact the Company where you try to verify your profile
```

**Comment by keeganquigley** [comment]
Hi @AurevoirXavier thanks for letting us know, still looking into this. Will let you know when I have an update.

**Comment by AurevoirXavier** [comment]
> Hi @AurevoirXavier thanks for letting us know, still looking into this. Will let you know when I have an update.

BTW, I have double identities. I can try using another one. Previously, I used the HongKong identity, which may not be supported.

**Comment by keeganquigley** [comment]
Hi @AurevoirXavier sure feel free to try the alternative one if you want, right now the current one is having issues being accepted.

**Comment by AurevoirXavier** [comment]
> Hi @AurevoirXavier sure feel free to try the alternative one if you want, right now the current one is having issues being accepted.

image

I attempted to use all my identities, including ID cards and passport. The identity document and selfie are passed. However, it consistently fails at the applicant data stage. I'm unsure of the reason. My legal name on the document is not my English name; it is presented in the local language. Does this have an impact? 

image

**Comment by PieWol** [comment]
Great, thanks.

**Review by PieWol — DISMISSED** [review]
I'm happy to approve it.

**Review by Noc2 — APPROVED** [review]

**Comment by keeganquigley** [comment]
Hi @AurevoirXavier I just sent you an email regarding KYB. I used the one you provided in your profile, so let me know if you receive it. Thanks!

**Comment by AurevoirXavier** [comment]
> Hi @AurevoirXavier I just sent you an email regarding KYC. I used the one you provided in your profile, so let me know if you receive it. Thanks!

Thanks, replied.

File: applications/polkadot-runtime-releaser.md
Comment (takahser) [inline applications/polkadot-runtime-releaser.md:29]:
```suggestion
    - I saw some EVM RPCs require the runtime version to be equal to the node version.
```

File: applications/polkadot-runtime-releaser.md
Comment (takahser) [inline applications/polkadot-runtime-releaser.md:30]:
```suggestion
- How to host a wasm-override repository for the community?
```

File: applications/polkadot-runtime-releaser.md
Comment (takahser) [inline applications/polkadot-runtime-releaser.md:66]: Could you structure this a bit better, so it's apparent which actions will be supported and what kind of use case they will cover?

For example:

| GitHub Action | Trigger  | Specification  |
| -----: | ------ | -------------- |
| `try-runtime` | by commenting `/bot try-runtime <network>` on the PR | when executed, XYZ will happen |

File: applications/polkadot-runtime-releaser.md
Comment (takahser) [inline applications/polkadot-runtime-releaser.md:130]: > It can build the specify feature WASM

Could you ellaborate on that?

> manage them well in the corresponding folder.

Does that mean that the build will be stored on the github repo?

File: applications/polkadot-runtime-releaser.md
Comment (takahser) [inline applications/polkadot-runtime-releaser.md:131]:
```suggestion
|     4. | WASM override CI              |  Listen for build requests from the release CI.                                                                             |
```

**Review by takahser — CHANGES_REQUESTED** [review]
Hi @AurevoirXavier 

Thanks for submitting this. I've added a couple of suggestions and comments, feel free to check it out.

**Comment by semuelle** [comment]
pinging @AurevoirXavier

**Comment by keeganquigley** [comment]
Closed as per discussions in private email thread.

**Comment by keeganquigley** [comment]
Re-opening per private email thread 😀

@AurevoirXavier can you address the above open comments? Thanks!

**Comment by AurevoirXavier** [comment]
> @AurevoirXavier can you address the above open comments? Thanks!

Absolutely!

**Comment by semuelle** [comment]
Are you still working on the changes and answers to [the comments above](https://github.com/w3f/Grants-Program/pull/2373#pullrequestreview-2326950526), @AurevoirXavier?

**Comment by AurevoirXavier** [comment]
> Are you still working on the changes and answers to [the comments above](https://github.com/w3f/Grants-Program/pull/2373#pullrequestreview-2326950526), @AurevoirXavier?

I was struggling with the KYC few days ago. And I'm off work thess two days. I'll finish this tomorrow. Sorry for the delay.

File: applications/polkadot-runtime-releaser.md
Comment (AurevoirXavier) [reply applications/polkadot-runtime-releaser.md:130]: > Does that mean that the build will be stored on the github repo?

Yes, that's pretty common. For example, https://github.com/moonbeam-foundation/moonbeam-runtime-overrides/tree/master/wasm.

Comment (AurevoirXavier) [reply applications/polkadot-runtime-releaser.md:130]: > Could you ellaborate on that?

Just as described, this is a versatile tool designed for compiling runtime with particular features. 

And I plan to utilize it for compiling the debug WASM and EVM tracing WASM within the GitHub Actions I previously referenced.

File: applications/polkadot-runtime-releaser.md
Comment (AurevoirXavier) [reply applications/polkadot-runtime-releaser.md]: Updated.

**Review by keeganquigley — DISMISSED** [review]
Thanks @AurevoirXavier re-approving

File: applications/polkadot-runtime-releaser.md
Comment (semuelle) [inline applications/polkadot-runtime-releaser.md:133]:
```suggestion
|    0b. | Documentation                 | There will be a guide to tell people how to use this and inline docs will cover core functionalities.                                                                 |
```

File: applications/polkadot-runtime-releaser.md
Comment (semuelle) [inline applications/polkadot-runtime-releaser.md:124]: I believe this is a copy-paste relict.

**Review by semuelle — APPROVED** [review]
Thanks for the update, happy to approve.

**Review by keeganquigley — APPROVED** [review]

**Review by Noc2 — APPROVED** [review]

**Pull request merged**
